lib/messagebird/http_client.rb

```diff
@@ -54,27 +54,17 @@ def request_block(method, path, params = {})
 
     def prepare_request(request, params = {})
       request.set_form_data(params)
+
       request
     end
 
     SUBMIT_METHODS = [:patch, :post, :put].freeze
 
     def build_request(method, uri, params = {})
       # Construct the HTTP request.
-      case method
-      when :delete
-        request = Net::HTTP::Delete.new(uri.request_uri)
-      when :get
-        request = Net::HTTP::Get.new(uri.request_uri)
-      when :patch
-        request = Net::HTTP::Patch.new(uri.request_uri)
-      when :post
-        request = Net::HTTP::Post.new(uri.request_uri)
-      when :put
-        request = Net::HTTP::Put.new(uri.request_uri)
-      else
-        raise MethodNotAllowedException
-      end
+      raise MethodNotAllowedException unless [:delete, :get, :patch, :post].include?(method)
```

Comment:
1. the change is not correct. missing :put method
2. the array needs be extracted to constant, suggested name `ALLOWED_METHODS`

Reply:
would be cool to have a simple unit test before making any change

Reply:
I will do it, I thought it already in the place

Reply:
the tests did not fail in your first push while it had missing :put method, so obviously no tests capture introduce change